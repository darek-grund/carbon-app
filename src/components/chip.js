import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Chip = ({ className, children, close }) => {
  return (
    <div className={ className }>
      {children}
      {close ? <span className='close' /> : false}
    </div>
  );
};


Chip.propTypes = {
  children: PropTypes.node,
  close: PropTypes.bool
};

Chip.defaultProps = {
  children: '',
  close: false
};

const StyledChip = styled(Chip)`
  display: inline-flex;
  height: 32px;
  align-items: center;
  border: 1px solid #ccc;
  border-radius: 16px;
  background: #e0e0e0;
  font-size: 13px;
  margin: 4px;

  padding: ${props => (props.close ? '0 4px 0 12px' : '0 12px')};

  .close {
    display: flex;
    width: 20px;
    height: 20px;
    background: #aaa;
    color: #fff;
    border-radius: 50%;
    margin: 0 2px 0 8px;
    cursor: pointer;
    align-items: center;
    justify-content: center;
    transition: .3s;
  }

  .close::after {
    content: 'âœ–';
    font-size: 10px;
  }

  .close:hover {
    background-color: #666;
  }
`;

export default StyledChip;
