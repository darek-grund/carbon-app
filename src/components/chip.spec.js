import React from 'react';
import { mount } from 'enzyme';
import { StyledChip } from './chip';

// Demonstration of a simple test for the HelloWorld component:
describe('<StyledChip />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<StyledChip />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render children when passed in', () => {
    wrapper.setProps({ children: <div>child element</div> });
    expect(wrapper.contains([<div>child element</div>])).toBe(true);
  });

  describe('close button', () => {
    let closeButton;

    beforeEach(() => {
      wrapper.setProps({ close: true });
      closeButton = wrapper.find('.close');
    });

    it('should render correctly', () => {
      expect(closeButton).toHaveLength(1);
    });

    it('should call function on click', () => {
      const onCloseCallback = jest.fn();
      wrapper.setProps({ onClose: onCloseCallback });
      closeButton.simulate('click');
      expect(onCloseCallback).toHaveBeenCalledTimes(1);
    });
  });
});
