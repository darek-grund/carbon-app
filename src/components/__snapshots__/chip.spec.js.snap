// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledChip /> should render correctly 1`] = `
<Styled(Chip)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "iXOmuq",
          "rules": Array [
            "
  display: inline-flex;
  height: 32px;
  align-items: center;
  border: 1px solid #ccc;
  border-radius: 16px;
  background: #e0e0e0;
  font-size: 13px;
  margin: 4px;
  padding: ",
            [Function],
            ";

  .close {
    display: flex;
    width: 20px;
    height: 20px;
    background: #aaa;
    color: #fff;
    border: none;
    border-radius: 50%;
    margin: 0 2px 0 8px;
    padding: 0;
    cursor: pointer;
    align-items: center;
    justify-content: center;
    outline: none;
    transition: .3s;

    &::after {
      content: 'âœ–';
      font-size: 10px;
    }

    &:hover {
      background-color: #666;
    }
  }
",
          ],
        },
        "displayName": "Styled(Chip)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
  >
    <Chip
      className="sc-bdVaJa iXOmuq"
      close={false}
    >
      <div
        className="sc-bdVaJa iXOmuq"
      />
    </Chip>
  </StyledComponent>
</Styled(Chip)>
`;
